site/src/pages/WorkspacePage/WorkspaceActions/DownloadLogsDialog.tsx

```diff
@@ -28,70 +36,124 @@ type DownloadableFile = {
 
 export const DownloadLogsDialog: FC<DownloadLogsDialogProps> = ({
   workspace,
+  open,
+  onClose,
   download = saveAs,
-  ...dialogProps
 }) => {
   const theme = useTheme();
-  const agents = selectAgents(workspace);
-  const agentLogResults = useQueries({
-    queries: agents.map((a) => ({
-      ...agentLogs(workspace.id, a.id),
-      enabled: dialogProps.open,
-    })),
-  });
+
   const buildLogsQuery = useQuery({
     ...buildLogs(workspace),
-    enabled: dialogProps.open,
+    enabled: open,
   });
-  const downloadableFiles: DownloadableFile[] = useMemo(() => {
-    const files: DownloadableFile[] = [
-      {
-        name: `${workspace.name}-build-logs.txt`,
-        blob: buildLogsQuery.data
-          ? new Blob([buildLogsQuery.data.map((l) => l.output).join("\n")], {
-              type: "text/plain",
-            })
-          : undefined,
-      },
+  const buildLogsFile = useMemo<DownloadableFile>(() => {
```

Comment:
Decided to split off the build log file generation into its own memoization step, because it's not likely to change all that often